### PR TITLE
Project mongodb Emelie Svensson

### PR DESCRIPTION
**Pull request by emeliesv**
This week I didn't reach for any stretch-goals due to feeling very tired.
I am, however, very pleased with the endresult of this API.

It fullfills the blue requirements, and I feel I comprehend what I did which is what matters.

https://project-goldenglobes-mongodb.herokuapp.com/

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,13 +1,28 @@
...
+In this API you can find categories for the Golden Globes and see all nominations for each category.
+You can also display all nominations if you'd like to.
 
-## The problem
+## The project
+This API has 5 routes.
+A minimum of one endpoint return a **collection** of results (array of elements)
+A minimum of one endpoint return a **single** result (single element).
+The API makes use of Mongoose models to model data and use these models to fetch data from the database.
+The API is [RESTful](https://www.smashingmagazine.com/2018/01/understanding-using-rest-api/)
```
Comment (Andrea-Osmar) [inline README.md:14]: Adding this link here is so smart :)

File: README.md [hunk 2]
```diff
@@ -1,13 +1,28 @@
...
+Javascript
+Mongoose
+MongoDB
 
 ## View it live
+<a href='https://project-goldenglobes-mongodb.herokuapp.com'>Project Golden Globes Mongo </a>
 
-Every project should be deployed somewhere. Be sure to include the link to the deployed project so that the viewer can click around and see what it's all about.
+## Author
+<a href='https://github.com/emeliesv'>Emelie Svensson</a>
```
Comment (Andrea-Osmar) [inline README.md:28]: Cool that you added the author 👍🏼

File: server.js [hunk 3]
```diff
@@ -2,37 +2,105 @@ import express from 'express'
...
+  year_award: Number,
+  ceremony: Number,
+  category: {
+    type: mongoose.Schema.Types.ObjectId,
+    ref: 'Category',
+  },
+  nominee: String,
+  film: String,
+  win: Boolean,
+})
```
Comment (Andrea-Osmar) [inline server.js:24]: Model looks good :) Similar to mine so is probably amazing ;)

File: server.js [hunk 4]
```diff
@@ -2,37 +2,105 @@ import express from 'express'
...
 
-// Start defining your routes here
 app.get('/', (req, res) => {
-  res.send('Hello world')
+  res.send(endpoints(app))
+})
+
+app.get('/nominations', async (req, res) => {
+  const movieNominations = await MovieNominee.find()
+  res.json(movieNominations)
```
Comment (Andrea-Osmar) [inline server.js:75]: very clean endpoint, good naming!

File: server.js [hunk 5]
```diff
@@ -2,37 +2,105 @@ import express from 'express'
...
+  const categories = await Category.find()
+  res.json(categories)
+})
+
+app.get('/categories/:id/', async (req, res) => {
+  const category = await Category.findById(req.params.id)
+  if (category) {
+    res.json(category)
+  } else {
+    res.status(404).json({ error: 'No category found' })
```
Comment (Andrea-Osmar) [inline server.js:88]: Very good error message which explains the error!

File: server.js [hunk 6]
```diff
@@ -2,37 +2,105 @@ import express from 'express'
...
+
+app.get('/categories/:id/nominations', async (req, res) => {
+  const category = await Category.findById(req.params.id)
+  if (category) {
+    const nominations = await MovieNominee.find({
+      category: mongoose.Types.ObjectId(category.id),
+    })
+    res.json(nominations)
+  } else {
+    res.status(404).json({ error: 'No nominations found' })
```
Comment (Andrea-Osmar) [inline server.js:100]: Same here, very good message!

File: server.js [hunk 7]
```diff
@@ -2,37 +2,105 @@ import express from 'express'
 import bodyParser from 'body-parser'
 import cors from 'cors'
 import mongoose from 'mongoose'
+import endpoints from 'express-list-endpoints'
```
Comment (Andrea-Osmar) [inline server.js:5]: I like that you are using the express-list-endpoints, it's very helpful!

File: server.js [hunk 8]
```diff
@@ -2,37 +2,105 @@ import express from 'express'
...
-// Defines the port the app will run on. Defaults to 8080, but can be 
-// overridden when starting the server. For example:
-//
-//   PORT=9000 npm start
+const MovieNominee = mongoose.model('MovieNominee', {
+  year_film: Number,
+  year_award: Number,
+  ceremony: Number,
+  category: {
+    type: mongoose.Schema.Types.ObjectId,
```
Comment (Andrea-Osmar) [inline server.js:18]: Cool combining here!

File: server.js [hunk 9]
```diff
@@ -2,37 +2,105 @@ import express from 'express'
...
+    await MovieNominee.deleteMany()
+
+    let categoryDocuments = []
+
+    const categories = goldenData.map(item => item.category)
+    const categorySet = new Set(categories)
+
+    categorySet.forEach(item => {
+      const category = new Category({category: item})
+      categoryDocuments.push(category)
```
Comment (Andrea-Osmar) [inline server.js:43]: Impressive that you went the extra step and added push!

**Review by Andrea-Osmar — COMMENTED** [review]
A very short review this time. 
You have done a great job! I like that you changed API from the express project as well. 
Your code is easy to read and your structure and indentation is flawless :D 
Personally I like some comments in the code to make it easier to follow but I guess it is not really necessary. 
You  reached all the blue requirements  and I think it is really good that you stopped after that and gave your self a break :) Rest is important!

**Review by KarolCislowski — APPROVED** [review]
Great Project 🥇

**Pull request closed without merging**